scripts/text_layers.jsx

```diff
@@ -270,12 +288,64 @@ var FormattedTextArea = Class({
      * A FormattedTextField where the text is required to fit within a given area. An instance of this class will step down the font size
      * until the text fits within the reference layer's bounds (in 0.25 pt increments), then rasterise the text layer, and centre it vertically
      * with respect to the reference layer's pixels.
+     * Positions the flavour text divider after sizing the text to the given area.
      */
 
     extends_: FormattedTextField,
-    constructor: function (layer, text_contents, text_colour, flavour_text, is_centred, reference_layer) {
+    constructor: function (layer, text_contents, text_colour, flavour_text, is_centred, reference_layer, divider_layer) {
         this.super(layer, text_contents, text_colour, flavour_text, is_centred);
         this.reference_layer = reference_layer;
+        this.divider_layer = divider_layer;
+    },
+    insert_divider: function() {
+        /**
+         * Inserts the flavour text divider between rules text and flavour text. 
+         * The position of the divider is calculated by creating two copies of `this.layer`, rendering `this.text_contents` in the first copy 
+         * and `this.flavour_text` in the second copy, then finding the midpoint between the bottom of the first copy and the top of the second copy.
+         * This method is slighty imperfect because of how the shape of the text layer can affect how flavour text is positioned, but should be close enough.
+         */
+
+        if (this.flavour_text.length !== "") {
```

Comment:
Might want to test either for length `if (this.flavour_text.length)` or for flavour text not being empty string `if (this.flavour_text !== ""`. As it stands now script seems to fail for cards with no flavour text.

Great work with this PR though 💪 super useful 🙏